src/systems/filecoin_mining/storage_mining/storage_miner_actor_code.go

```diff
@@ -605,7 +593,15 @@ func (a *StorageMinerActorCode_I) ProveCommitSector(rt Runtime, info sector.Sect
 		// expired
 		delete(st.PreCommittedSectors(), preCommitSector.Info().SectorNumber())
 		UpdateRelease(rt, h, st)
-		return rt.ErrorReturn(exitcode.UserDefinedError(0)) // TODO: user dfined error code?
+
+		TODO()
+		// TODO: if we abort, the above state changes (delete...) will not be committed.
+		// Is this desired? (alternative: support method failure while still committing
+		// state changes if UpdateRelease has already been called)
```

Comment:
no I believe they should be committed (cc @zixuanzh). Does `UpdateRelease` get aborted?

Reply:
This is that one case I identified earlier where an actor is attempting to do some clean up even while the method fails. We should rework the code so that this doesn't depend on clean-up happening here. (It's not clear to me that anything does depend on this - it might be ok just to remove the code).

Reply:
Alternatively, we could introduce a variant of Abort that does this (e.g., AbortPreservingUpdates vs. AbortDiscardingUpdates)?